ClassAttendance/src/StudentApi/Controllers/ClassesController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Threading.Tasks;
+using BLL.Exceptions;
+using BLL.Interfaces;
+using BLL.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace ClassAttendanceAPI.Controllers
+{
+    [Route("[controller]")]
+    public class ClassesController : Controller
+    {
+        private readonly IService<Class> _classService;
+        private readonly ILogger<ClassesController> _logger;
+
+        public ClassesController(IService<Class> classService, ILogger<ClassesController> logger)
+        {
+            _classService = classService;
+            _logger = logger;
+        }
+
+        [HttpGet("Classes")]
+        public IActionResult GetAll()
+        {
+            var result = _classService.GetAll();
+            _logger.LogInformation("Fetched classes");
+
+            return Ok(result);
+        }
+
+        [HttpGet("Class")]
```

Comment:
To not repeat the "Class" word, I advise you to not change the route at all.
In this case, you will have the next endpoints:
GET: api/v1/Classes
GET: api/v1/Classes/{id}
POST: api/v1/Classes (with body)
DELETE: api/v1/Classes/{id}
PUT: api/v1/Classes (with body)

Make the save for all controllers.
PD For `v1` read more about AddVersioning (Startup)

Reply:
Got it, i'll try to complete this and finish it ASAP.

Reply:
Done